src/lxml/includes/etree_defs.h

```diff
@@ -247,6 +247,22 @@ long _ftol2( double dblSource ) { return _ftol( dblSource ); }
 #define _isString(obj)   (PyUnicode_Check(obj) || PyBytes_Check(obj))
 #endif
 
+#if PY_MAJOR_VERSION < 3
+#define _isStringOrPathLike(obj)   (PyString_CheckExact(obj)  || \
+                                    PyUnicode_CheckExact(obj) || \
+                                    PyType_IsSubtype(Py_TYPE(obj), &PyBaseString_Type))
+#else
+#if PY_MINOR_VERSION < 6
```

Comment:
This is going to fail when Python 4.0 comes out. Use `PY_VERSION_HEX  < 0x03060000` instead.
Also, this seems to repeat what `_isString()` already does. Just use that.